pkg/api/persistentvolumeclaim/util_test.go

```diff
@@ -74,52 +74,48 @@ func TestDropDisabledSnapshotDataSource(t *testing.T) {
 	// Ensure that any data sources aren't enabled for this test
 	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.AnyVolumeDataSource, false)()
 
-	for _, enabled := range []bool{true, false} {
-		for _, oldpvcInfo := range pvcInfo {
-			for _, newpvcInfo := range pvcInfo {
-				oldPvcHasDataSource, oldpvc := oldpvcInfo.hasDataSource, oldpvcInfo.pvc()
-				newPvcHasDataSource, newpvc := newpvcInfo.hasDataSource, newpvcInfo.pvc()
-				if newpvc == nil {
-					continue
+	for _, oldpvcInfo := range pvcInfo {
+		for _, newpvcInfo := range pvcInfo {
+			oldPvcHasDataSource, oldpvc := oldpvcInfo.hasDataSource, oldpvcInfo.pvc()
+			newPvcHasDataSource, newpvc := newpvcInfo.hasDataSource, newpvcInfo.pvc()
+			if newpvc == nil {
+				continue
+			}
+
+			t.Run(fmt.Sprintf("old pvc %v, new pvc %v", oldpvcInfo.description, newpvcInfo.description), func(t *testing.T) {
+				var oldpvcSpec *core.PersistentVolumeClaimSpec
+				if oldpvc != nil {
+					oldpvcSpec = &oldpvc.Spec
 				}
+				DropDisabledFields(&newpvc.Spec, oldpvcSpec)
 
-				t.Run(fmt.Sprintf("feature enabled=%v, old pvc %v, new pvc %v", enabled, oldpvcInfo.description, newpvcInfo.description), func(t *testing.T) {
-					defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.VolumeSnapshotDataSource, enabled)()
+				// old pvc should never be changed
+				if !reflect.DeepEqual(oldpvc, oldpvcInfo.pvc()) {
+					t.Errorf("old pvc changed: %v", diff.ObjectReflectDiff(oldpvc, oldpvcInfo.pvc()))
+				}
 
-					var oldpvcSpec *core.PersistentVolumeClaimSpec
-					if oldpvc != nil {
-						oldpvcSpec = &oldpvc.Spec
+				switch {
```

Comment:
Can we simplify this test and get rid of all the case statements? Every case does the same thing now

Reply:
Done